local/src/server/verifier-server.cc

```diff
@@ -347,24 +376,49 @@ ServerReplayFileHandler::txn_close()
 }
 
 void
-TF_Serve(std::thread *t)
+delete_thread_info_session(ServerThreadInfo &thread_info)
+{
+  if (thread_info._session == nullptr) {
+    return;
+  }
+  std::unique_lock<std::mutex> lock(thread_info._mutex);
+  delete thread_info._session;
+  thread_info._session = nullptr;
+}
+
+void
+TF_Serve_Connection(std::thread *t)
 {
   ServerThreadInfo thread_info;
   thread_info._thread = t;
   while (!Shutdown_Flag) {
     swoc::Errata errata;
 
     Server_Thread_Pool.wait_for_work(&thread_info);
+    if (Shutdown_Flag) {
+      // Calling Shutdown is a condition that releases wait_for_work.
+      delete_thread_info_session(thread_info);
+      break;
+    }
 
     errata = thread_info._session->accept();
-    while (!thread_info._session->is_closed() && errata.is_ok()) {
+    while (!Shutdown_Flag && !thread_info._session->is_closed() && errata.is_ok()) {
       HttpHeader req_hdr;
       swoc::Errata thread_errata;
       swoc::LocalBufferWriter<MAX_HDR_SIZE> w;
+      auto &&[poll_return, poll_errata] = thread_info._session->poll(Thread_Sleep_Interval_MS);
+      if (poll_return == 0) {
+        // Poll timed out. Loop back around.
+        continue;
+      } else if (poll_return < 1) {
```

Comment:
Better to check `poll_erata.is_ok()` - it's just as fast and a big point of using `Rv` is to not have "special" cases of the base value to indicate failure.